gui/mozregui/main.py

```diff
@@ -97,11 +98,33 @@ def show_about(self):
     def edit_global_prefs(self):
         change_prefs_dialog(self)
 
+    @Slot(str)
+    def on_log_received(self, log_message):
+        self.ui.log_information_view.appendPlainText(log_message)
+        counter = self.ui.log_information_view.blockCount()
+        print "counter is ", counter
+        if(counter > 1000):
+            self.ui.log_information_view.clear()
+
+
+class LogModel(QObject):
+    log = Signal(str)
+
+    def __init__(self):
+        QObject.__init__(self)
+
+    def __call__(self, data):
+        print "I got a log", data
+        log_message = '{time:d}:{level:s}:{message:s}'.format(**data)
+        print "I got a message", log_message
+        self.log.emit(log_message)
```

Comment:
I would prefer that we send the raw log dict here. Then the QPlainTextEdit will do the formatting.